src/py42/_internal/clients/alertrules.py

```diff
@@ -0,0 +1,161 @@
+import json
+from py42.clients import BaseClient
+from py42.clients.util import get_all_pages
+
+
+_ALERT_RULE_ENDPOINTS = {
+    u"exfiltration": u"query-endpoint-exfiltration-rule",
+    u"cloudshare": u"query-cloud-share-permissions-rule",
+    u"typemismatch": u"query-file-type-mismatch-rule",
+}
+
+
+class AlertRulesClient(BaseClient):
+    """A client to manage Alert Rules."""
+
+    _version = u"v1"
+    _resource = u"Rules/"
+    _api_prefix = u"/svc/api/{0}/{1}".format(_version, _resource)
+
+    def __init__(self, session, user_context):
+        super(AlertRulesClient, self).__init__(session)
+        self._user_context = user_context
+
+    def add_user(self, rule_id, user_id, user_aliases):
+        """Update alert rule criteria to apply the rule for specific user(s) for the given rule id.
+
+        Args:
+            rule_id (str): Id of a rule to be updated.
+            user_id (str): User id who needs to be added to the criteria.
+            user_aliases (str): Email ids/Aliases who needs to be added to criteria.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        tenant_id = self._user_context.get_current_tenant_id()
+        data = {
+            u"tenantId": tenant_id,
+            u"ruleId": rule_id,
+            u"userList": [
+                {
+                    # not sure what should be passed here and its significance.
```

Comment:
I need to confirm the behaviour and difference between the usages of both the fields, whether one of them is optional, whether they are mutually exclusive, etc. I will do some testing and play around a bit here, right now just added as is since the API expects it. 
This is pending as I need to figure out the key to connect to `fed-observer` instance